### PR TITLE
Virt cdrom fix

### LINKED ISSUE #57477: virt.init doesn't generate proper disk target devices
### Description of Issue

When generating the XML for a VM that has a mix of disk buses using`virt.init`, the disk target names are not all starting from `a`.

For instance with the following disks is generating the following targets: `vda` and `hdb` instead of `vda` and `hda`.

```python
[
   {"name": "system", "pool": "default"},
    {
        "name": "tested",
        "device": "cdrom",
        "source_file": None,
        "model": "ide",
    },
]
```

### PR DESCRIPTION
**Pull request by cbosdo**
### What does this PR do?

This issue fixes some issues with the handling of CDROM images attaching and changing on VM created or updated by `virt.init()` and `virt.update()` functions.

### What issues does this PR fix or reference?
Fixes: #57477

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [X] Docs
- [X] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by twangboy — DISMISSED** [review]

**Review by waynew — DISMISSED** [review]

**Comment by waynew** [comment]
Pretty sure this will get merged after the code thaw :ice_cube: :sun_with_face:

**Comment by cbosdo** [comment]
> Pretty sure this will get merged after the code thaw ice_cube sun_with_face

you mean it's likely not to hit Sodium?

**Review by Akm0d — DISMISSED** [review]

**Comment by cbosdo** [comment]
Rebased on master, added more tests and fixed an uncovered bug in the target name generation code

**Comment by cbosdo** [comment]
rebased on master

**Comment by cbosdo** [comment]
Updated on latest master

File: salt/modules/virt.py [hunk 1]
```diff
@@ -2163,6 +2167,21 @@ def _remove_indent(node):
...
+
+    :param targets: the list of already computed targets
+    :param disks: the number of disks
+    :param prefix: the prefix of the target name, i.e. "hd"
+    """
+    return [
+        "{0}{1}".format(prefix, string.ascii_lowercase[i])
+        for i in range(disks_count)
+        if "{0}{1}".format(prefix, string.ascii_lowercase[i]) not in targets
+    ][0]
```
Comment (DmitryKuzmenko) [inline salt/modules/virt.py:2182]: Looks good but logically it should be more effective if we'll not build the entire list to just get the first item. Maybe something like this?
```
for i in range(disks_count):
    ret = "{0}{1}".format(prefix, string.ascii_lowercase[i])
    if ret not in targets:
        return ret
```

Comment (cbosdo) [reply salt/modules/virt.py:2182]: right, I'll change that then

**Comment by cbosdo** [comment]
@DmitryKuzmenko Just included your suggestion

**Review by DmitryKuzmenko — DISMISSED** [review]

**Review by Akm0d — DISMISSED** [review]

**Comment by cbosdo** [comment]
rebased on master

**Review by Akm0d — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**